python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -1075,6 +1075,81 @@ async def move_rotator(self, position: float) -> None:
             component_name="MTRotator",
         )
 
+    async def move_m1m3_hp(self, hp_index: int, step_size: int) -> None:
+        """Move a single M1M3 hardpoint (hp) to a new position
+
+        Parameters
+        ----------
+        hp_index: `int`
+            Hardpoints index (from 0 to 5).
+        """
+        steps = [0] * 6
+        steps[hp_index] = step_size
+
+        self.log.info(f"Moving Hardpoint {hp_index} in {step_size} steps - Start")
+        await self.rem.mtm1m3.cmd_moveHardpointActuators.set_start(
+            steps=steps, timeout=self.long_timeout
+        )
+
+        # Wait a bit
+        await asyncio.sleep(1)
```

Comment:
Why do you need to "wait a bit"?

Reply:
This is poor man response to the fact that at least some version of hardpoint controller did not send updates before command was completed. So the script will get "hardpoint in idle" when looking for hardpoint state, decide it is and go to next steps - it will miss transition into "hardpoint stepping" which occurs when HP starts to move.

We can fix that (if not done already, but I doubt that) on CSC side, to prevent command return earlier then HP moves a bit. Or we can improve script by quering for a state, but ignoring IDLE state for a while.

Changing CSC behaviour isn't easy, as that requires changes into how command are processed. Due to simplification, commands are taken from a FIFO, one of the "pseudo" command is update/ILC pool, which runs every 20ms (50 Hz). So to fix that, I will need to introduce new command, "check HP movement completion" (or at least methods to the Command class handling that). Likely doable, but we shall start another DM-ticket to fix that.

All this pertains to the original idea "M1M3 commands are short-lived, all is reported back as events/status updates". We know it is bad.

Reply:
ok. that's fair.. I think then, the right thing to do is give the CSC a bit of time for this to occur.. But, instead of "wait a bit", we can handle it in a more robust way.. getting `next` samples and checking that they go to the expected value before a timeout occurs..